textx/lang.py

```diff
@@ -89,7 +89,7 @@ def comment_block():        return _(r'/\*(.|\n)*?\*/')
 ID          = _(r'[^\d\W]\w*\b', rule_name='ID', root=True)
 BOOL        = _(r'(True|true|False|false|0|1)\b', rule_name='BOOL', root=True)
 INT         = _(r'[-+]?[0-9]+\b', rule_name='INT', root=True)
-FLOAT       = _(r'[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?\b', 'FLOAT',
+FLOAT       = _(r'[+-]?(\d+(\.\d*)?|\.\d+)([eE][+-]?\d+)?', 'FLOAT',
```

Comment:
What's the reason behind removing `\b` at the end? There are some subtle gotchas if anchoring is not used. I made an extension to float test in cf22658 that passes with original textX but fails with this extension. The problem is when you have something that matches FLOAT as a prefix of some other match.

Reply:
a simple \b would not be sufficient as the literal `.` doesn't count as a word boundary - I've added a custom boundary that behaves just like \b but looks for a word char /not word char transition including `.` as a word character.